core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -4565,4 +4642,16 @@ public enum FlatProductInputType {
     SCALAR, LIST, MAP
   }
 
+  /** Type of part to extract passed into {@link #parseUrl}. */
+  public enum PartToExtract {
+    HOST,
```

Comment:
Is there a reason for this specific ordering? If not maybe we could alphabetize this and the switch statement? This is just a nit so if you feel strongly one way or the other it's probably fine.

Reply:
IMO the enum type doesn't need to keep alphabetize order, like `SqlKind`.
In this case, this is related to the order of the extracted parts of URL, like HOST/PATH/QUERY/REF. Although the latter part is not strictly in accordance with the order of the URL, it may be used by users less, so it is placed later.
And this way is keep the same as Hive's and Spark's implementation, so I think it's probably fine.